src/sentry/integrations/source_code_management/repository.py

```diff
@@ -0,0 +1,213 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Collection, Mapping, Sequence
+from typing import Any
+
+import sentry_sdk
+
+from sentry.auth.exceptions import IdentityNotValid
+from sentry.integrations.services.integration.model import RpcIntegration
+from sentry.integrations.services.repository import RpcRepository
+from sentry.models.identity import Identity
+from sentry.models.integrations.integration import Integration
+from sentry.models.repository import Repository
+from sentry.shared_integrations.client.base import BaseApiResponseX
+from sentry.shared_integrations.exceptions import ApiError, IntegrationError
+from sentry.utils import metrics
+
+REPOSITORY_INTEGRATION_CHECK_FILE_METRIC = "repository_integration.check_file.{result}"
+REPOSITORY_INTEGRATION_GET_FILE_METRIC = "repository_integration.get_file.{result}"
+
+
+class RepositoryIntegration(ABC):
+    repo_search = False
+    model: RpcIntegration | Integration
```

Comment:
`model` should also be an abstract property.

For consistency's sake, let's go ahead and make `repo_search` a property as well (but a non-abstract one that returns `False` unless overridden). The class-level variable works for now, but I like a property method because (1) it allows subclasses to depend on instance state if they want to and (2) it's a better place to put a docstring.

Reply:
Afterthought: unless we care about reading those attributes directly from subclass type tokens, not from instances. But even in that case I think it may be possible to use properties with `@classmethod`. LMK if you want help with that, or else I'm fine just leaving them as class variables if it saves trouble.

Reply:
`repo_search` looks to always be `True` in the installation 🤔

Reply:
@RyanSkonnord `model` is also set, should i also implement a setter for it?